### ISSUE TITLE
Susceptibility Distortion Correction - effective echo-spacing and total-readout time for Philips data

### ISSUE THREAD

**Issue by treanus** [nipreps/sdcflows#5]
Dear all,

First, thank you for providing fmriprep - very useful!

I have questions about ESS and TRT for SDC of bold fMRI with a B0 fieldmap using Philips data.

In the documentation of fmriprep SDC and source code of fmriprep.interfaces.fmap there are the functions get_ees and get_trt. For Philips data it seems to be enough to define the WaterFatShift, MagneticFieldStrength, PhaseEncodingDirection and ParallelReductionFactorInPlane to calculate ess and trt.

**First question:**
a) does this signify that the EffectiveEchoSpacing does not need to be specified in the BIDS json file of the bold fmri, when WaterFatShift, MagneticFieldStrength, PhaseEncodingDirection and ParallelReductionFactorInPlane are given in the json file?
b) this then gives a warning in the bids-validation step ("You should define 'EffectiveEchoSpacing' for this file. If you don't provide this information field map correction will not be possible.") that can be safely ignored?

**Second Question:**
a) looking at the source code:
```
-
if wfs is not None:
        fstrength = in_meta['MagneticFieldStrength']
        wfd_ppm = 3.4  # water-fat diff in ppm
        g_ratio_mhz_t = 42.57  # gyromagnetic ratio for proton (1H) in MHz/T
        wfs_hz = fstrength * wfd_ppm * g_ratio_mhz_t
        return wfs / (wfs_hz * etl)
-
```
it seems that ParallelReductionFactorInPlane (the sense acceleration) is not used?

b) this ParallelReductionFactorInPlane is not used, since the echo-train-length (etl) in Philips images is already taking this into account? Correct?

c) I'm confused by other information on the web (among others: http://www.spinozacentre.nl/wiki/index.php/NeuroWiki:Current_developments#B0_correction) that also provide a calculation for the ESS as:
effective echo spacing = (((1000 * wfs)/(434.215 * (EPI factor+1))/acceleration)

Thus here 'acceleration' is used. Acceleration means the sense factor, which is also the same as the ParallelReductionFactorInPlane? Correct?
I guess it all depends on how 'etl' is defined. In the philips definition it already incorporates the ParallelReductionFactorInPlane (etl =  NumberOfPhaseEncodingSteps/ParallelReductionFactorInPlane)

**Last Question:**

Starting from Philips dicom, I need to extract the correct information to put into the BIDS json file.

Using matlab dicominfo to extract relevant information out of Philips Dicom, one finds the following:
- MagneticFieldStrength in 'MagneticFieldStrength'
- etl in 'Private_2001_1013'
- WaterFatShift in 'Private_2001_1022'
- ParallelReductionFactorInPlane in 'Private_2005_140f.Item_1.ParallelReductionFactorInPlane';

EffectiveEchoSpacing is not in the philips dicom header, and is thus calculated.

Hope someone can help with (part of) my questions...

Best regards,

Stefan

Ps. Slightly of topic:

The PhaseEncodingDirection is not fully specified in the philips dicomheader? One cannot find e.g. 'j' or 'j-' as both are defined as 'COL' in the philips dicom header?
Also the slice timing info is not in the philips dicom header?
However ParallelReductionFactorOutOfPlane is specified in the philips dicom header, and this corresponds to the multiband factor?

**Comment by oesteban** [comment]
One more interesting resource (recommended after a revision of the FSL FUGUE user guide) when we check on this - https://osf.io/hks7x/?show=view

**Comment by effigies** [comment]
This seems like the relevant issue, but I can open a new one if it's sufficiently different. I've recently converted some Philips phase1/phase2 fieldmaps with dcm2niix and gotten the following sidecars:

```JSON
{
        "Manufacturer": "Philips",
        "PatientPosition": "HFS",
        "SeriesDescription": "ImageMRSERIES",
        "ProtocolName": "FieldMapping",
        "SeriesNumber": 16,
        "AcquisitionNumber": 16,
        "ImageComments": "Audio",
        "PhilipsRescaleSlope": 1.53455,
        "PhilipsRescaleIntercept": -3142,
        "PhilipsScaleSlope": 651.74,
        "UsePhilipsFloatNotDisplayScaling": 1,
        "EchoNumber": 1,
        "EchoTime": 0.0023,
        "RepetitionTime": 0.02,
        "ReconMatrixPE": 256,
        "ImageOrientationPatientDICOM": [
                1,
                0,
                0,
                0,
                0.996551,
                -0.0829821      ],
        "ConversionSoftware": "dcm2niix",
        "ConversionSoftwareVersion": "v1.0.20190902"
}
```

```JSON
{
        "Manufacturer": "Philips",
        "PatientPosition": "HFS",
        "SeriesDescription": "ImageMRSERIES",
        "ProtocolName": "FieldMapping",
        "SeriesNumber": 16,
        "AcquisitionNumber": 16,
        "ImageComments": "Audio",
        "PhilipsRescaleSlope": 1.53455,
        "PhilipsRescaleIntercept": -3142,
        "PhilipsScaleSlope": 651.74,
        "UsePhilipsFloatNotDisplayScaling": 1,
        "EchoNumber": 2,
        "EchoTime": 0.0046,
        "RepetitionTime": 0.02,
        "ReconMatrixPE": 256,
        "ImageOrientationPatientDICOM": [
                1,
                0,
                0,
                0,
                0.996551,
                -0.0829821      ],
        "ConversionSoftware": "dcm2niix",
        "ConversionSoftwareVersion": "v1.0.20190902"
}
```

None of the methods for recovering EES seem to work with this metadata.

I'll also ping @neurolabusc in case there's something I need to do to finagle dcm2niix to retrieve the needed metadata.

For reference, the current `get_ees` code is:

https://github.com/poldracklab/sdcflows/blob/03ce021a792ee5a9ba23f99c7f9ddc3afcd494f2/sdcflows/interfaces/fmap.py#L382-L462

Happy to share data privately, but waiting for IRB approval to make public.

**Comment by neurolabusc** [comment]
Chris,

1.  Your minimal BIDS sidecar for the phase map reflects the fact that Philips PAR/REC files are devoid of virtually all meta information. This is a limitation of the input data not dcm2niix.

I would urge all Philips users to save DICOM data direct from the console, the PAR/REC format is too lean to be considered archival quality. As @drmclem [noted](https://github.com/rordenlab/dcm2niix/issues/255): `the PAR-REC format was built for simpler times and its always been a tension between simplicity and comprehensiveness of the format.`

2. With regards to the @treanus comments, it is worth noting that Philips DICOMs have evolved a lot over the years. I have tried to get as much information as possible, and some details (like phase encoding polarity and slice timing) are simply not in the Philips DICOM. I do not have access to Philips hardware, but if someone wants to develop a validation dataset with modern images that validate robust effective readout time estimates, I am happy to extend dcm2niix. As has been noted [before](https://github.com/rordenlab/dcm2niix/issues/308), the "correct" value for TotalReadoutTime is only important in the context of FSL's topup if you need/want calibrated field maps, OR if for some reason your two acquisitions with differing polarities have differing TotalReadoutTimes. In most situations you can supply a bogus value, and the images used for subsequent processing are identical to if you had the precise values.

**Comment by effigies** [comment]
Thanks for weighing in, Chris. I need to integrate all of this a bit. Hopefully these fieldmaps are salvageable, but the ship has sailed in terms of getting DICOMs with the metadata we're currently expecting.

**Comment by neurolabusc** [comment]
@treanus I have updated dcm2niix to use the DICOM tags to populate the following BIDS tags:

```
	"ParallelReductionFactorInPlane": 2.5,
	"WaterFatShift": 7.95575,
	"EffectiveEchoSpacing": 0.000591038,
```

I used [these samples](https://github.com/neurolabusc/dcm_qa_philips) to develop this. Since I do not have Philips data, I would appreciate if others would test this. Also, tell me if `(wfs_hz * etl)` or `(wfs_hz * etl+1)` is preferred. As noted, EffectiveEchoSpacing typically has no impact, but it is nice to include for completeness if we can.

Note that until Philips specifies the phase encoding polarity, dcm2niix will only be able to populate `PhaseEncodingAxis` (e.g. `i` or `j`)rather than `PhaseEncodingDirection` (disambiguating `j` from `j-`).

**Comment by oesteban** [comment]
> Note that until Philips specifies the phase encoding polarity, dcm2niix will only be able to populate `PhaseEncodingAxis` (e.g. `i` or `j`)rather than `PhaseEncodingDirection` (disambiguating `j` from `j-`).

wow, that's really unfortunate!

**Comment by neurolabusc** [comment]
We do not have Philips hardware at my center, but it might be nice for someone with access to Philips equipment (@baxpr?) to acquire a validation dataset similar to what @mharms acquired on for [Siemens](https://github.com/neurolabusc/dcm_qa/tree/master/In/TotalReadoutTime), see his Excel spreadsheet for more details.

Glancing at your formula, I wonder if partial Fourier is handled correctly. As I recall, when calculating effective echo spacing using echo train length you ignore in plane acceleration (as that covers all of K-space, e.g. with SENSE=2 a 80 row scan with ETL=40 has the same distortion as an unaccelerated 40 row scan with the same FOV), but you must adjust based on partial Fourier (e.g. for a pF=5/8 with a 80-row scan and ETL=50 one has the same distortion as an unaccelerated 80 row scan). 

Therefore, I think your formula needs to use the ParallelReductionFactorInPlane to determine if the scan has partial Fourier (or have some other measure to detect pF) and adjust your calculation.

Again, since in typical cases effective echo spacing has no impact on subsequent processing, one may ignore this. However, I hate adding new features to my software when I know that they make assumptions that are not always correct.

**Comment by baxpr** [comment]
@neurolabusc this question has come up here, too. I'll add this to my list - hopefully we can get some useful info or test scans.

**Comment by neurolabusc** [comment]
@treanus, @baxpr, @effigies and @oesteban I would be grateful if you could look at the formula and Excel spreadsheet provided [here](https://github.com/rordenlab/dcm2niix/issues/377).

@baxpr, no need for test scans, as I have some from two sites, with permission to share from at least one site.

**Comment by oesteban** [comment]
Addressed in #132 - accepting patches on the `dev/1.4.0` branch.
